internal/provider/hypercore_disk_resource.go

```diff
@@ -164,6 +178,13 @@ func (r *HypercoreDiskResource) Create(ctx context.Context, req resource.CreateR
 				"capacity":      originalVDSizeBytes,
 			},
 		}
+
+		if isAttachingISO {
```

Comment:
This `if isAttachingISO` is inside `if sourceVirtualDiskID != ""` - this combination does not make sense, if you are cloning a VirtualDisk, then you do not have CD-ROM. This nested if must be dead code, you can remove it. Correct?

Reply:
Removed